### DISCUSSION TITLE
Errors in checkoutComplete cause payments to be CAPTURED with order=None and checkout=None

### DISCUSSION THREAD

**Discussion by sfehlandt** [saleor/saleor#9486] [Q&A]
Hi,

I am having an issue where I find some customers that pay through a payment gateway with the 2-step process (authorization and capture).
The payment ends CAPTURED but the references of the payment to both checkout and order are null.

We are trying to debug this and we certainly have custom payment gateways so I cannot ensure if the problem is from Saleor or if its on our end. However, from my understanding this condition should be impossible in Saleor, regardless of whatever error could have happened in a given plugin, right?

¿Has someone experienced something similar?
To give you some context, we are running Saleor in a kubernetes cluster with many single-worker pods running saleor (roughly 30).

Kind regards

**Comment by karolkielecki** [comment]
Hi @sfehlandt, could you please share more info? What's your Saleor version?

**Reply by sfehlandt** [reply]
Hi @karolkielecki sorry forgot about that:

We are using Saleor 3.0.4
We have a custom plugin that overwrites the `calculate_checkout_shipping`, `calculate_checkout_total`, `calculate_checkout_line_total` and `calculate_checkout_line_unit_price` methods.

We also have custom payment gateways.
Unfortunately I don't have logs because we do not have a centralized log collector, and since we use kubernetes pods with autoscalling, I cannot get logs for the particular cases where this has happened. Also, I haven't been able to reproduce this in my development environment.

We have a couple of payment gateways implementations where we receive (on a backend service) notification of the payment by the payment provider. In those cases, if the client has not come back yet to our site (and hence, we have not yet performed the second `checkoutComplete` mutation), when we receive the notification, we perform the second `checkoutComplete` ourselves (from the backend service that received the notification). So I am thinking if maybe this issue could be caused by a race condition between 2 `checkoutComplete` calls handled by 2 different saleor instances (k8s pods)?

Kind regards
